src/main/java/com/synopsys/integration/alert/common/digest/filter/NotificationFilter.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    private Set<String> getConfiguredNotificationTypes(final List<CommonDistributionConfig> distributionConfigs) {
+        return distributionConfigs
+                   .parallelStream()
+                   .flatMap(config -> config.getNotificationTypes().stream())
+                   .collect(Collectors.toSet());
+    }
+
+    private Map<String, List<NotificationContent>> getNotificationsByType(final Set<String> notificationTypes, final Collection<NotificationContent> notifications) {
+        final Map<String, List<NotificationContent>> notificationsByType = new HashMap<>();
+        notificationTypes.parallelStream().forEach(type -> {
+            final List<NotificationContent> applicableNotifications = notifications
+                                                                          .stream()
+                                                                          .filter(notification -> type.equals(notification.getNotificationType()))
+                                                                          .collect(Collectors.toList());
+            notificationsByType.put(type, applicableNotifications);
+        });
+        return notificationsByType;
+    }
+
+    public List<ProviderContentType> getProviderContentTypes() {
+        return providerDescriptors
+                   .parallelStream()
+                   .flatMap(descriptor -> descriptor.getProviderContentTypes().stream())
+                   .collect(Collectors.toList());
+    }
+
+    public Set<HierarchicalField> getFilterableFieldsByNotificationType(final String notificationType, final List<ProviderContentType> contentTypes) {
+        return contentTypes
+                   .parallelStream()
+                   .filter(contentType -> notificationType.equals(contentType.getNotificationType()))
+                   .flatMap(contentType -> contentType.getFilterableFields().stream())
+                   .collect(Collectors.toSet());
+    }
+
+    private Predicate<NotificationContent> createFilter(Collection<HierarchicalField> filterableFields, final Collection<CommonDistributionConfig> distributionConfigs) {
+        JsonFilterBuilder filterBuilder = DefaultFilterBuilders.ALWAYS_TRUE;
+        for (final CommonDistributionConfig config : distributionConfigs) {
+            for (final HierarchicalField field : filterableFields) {
+                Collection<String> valuesFromField = getValuesFromConfig(field.getConfigNameMapping(), config);
+                if (valuesFromField != null) {
+                    final JsonFilterBuilder fieldFilter = createFilterBuilderForAllValues(field, valuesFromField);
+                    filterBuilder = new AndFieldFilterBuilder(filterBuilder, fieldFilter);
+                }
+            }
+        }
+        return filterBuilder.buildPredicate();
+    }
+
+    private Collection<String> getValuesFromConfig(final String configKey, final CommonDistributionConfig config) {
```

Comment:
I think for the time being we want to leave it here because this will probably change significantly with the dynamic UI updates. Once we have a long term, maintainable way to do this, then we can abstract it.